### PR TITLE
Fix rocket v0.5.0-rc2 mitchmatch.

### PR DESCRIPTION
**Pull request by viljami**
Prior this PR there were bunch of compilation errors for `cargo test`. Some minor changes in rc2 caused them. 

Maybe the UUID related code could be in another PR. But without it the tests will not run. That part of the code should be revisited. 

I am also happy if this PR spawns another one which fixes the issues. It is completely ok to abandon this one.

### CODE REVIEW

**Review by blitz — COMMENTED** [review]
It would be good to get this merged in some form to unblock updates to the new rocket version. :)

File: examples/uuid/src/uuid.rs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    fn json_schema(_gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        schemars::schema::Schema::Object(schemars::schema::SchemaObject {
+            ..Default::default()
+        })
+    }
+}
+
+// impl<'r> OpenApiFromParam<'r> for Uuid {
```
[the remark this answers is not in the document]
Comment (viljami) [reply examples/uuid/src/uuid.rs:55]: Sorry, seems I left it in. I'll remove the commented code.

File: examples/uuid/src/uuid.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
+pub extern crate uuid as uuid_crate;
```
[the remark this answers is not in the document]
Comment (viljami) [reply examples/uuid/src/uuid.rs:1]: I am not entirely sure if this is the most suitable fix. I tried to counter the following error:

```
  --> examples/uuid/src/main.rs:43:1
   |
43 | #[openapi(tag = "Users")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `FromParam<'_>` is not implemented for `uuid::Uuid`
   |
   ```
   
   And the rabbit hole went deep.

[the remark this answers is not in the document]
Comment (viljami) [reply examples/uuid/src/uuid.rs:1]: I could not write `impl` for uuid crate. And I had to write a local struct to counter the limitation.

**Comment by viljami** [comment]
Commented and updated the description. Is the PR still useful? :)

**Comment by ralpha** [comment]
Thanks for the PR, but a lot of changes overlap with #89.
The UUID problem has since been solves as `Schemars` had an update and we could update it to `v1.x`.
I'll close this PR as most of the changes are merged when #89 was merged.

**Pull request closed without merging**